soroban-env-host/Cargo.toml

```diff
@@ -12,6 +12,10 @@ rust-version = "1.74"
 build = "build.rs"
 
 [dependencies]
+# To be removed
+anyhow = { version = "1.0.75", features = [] }
+thiserror = "1.0.46"
```

Comment:
I understand this needs to go, I want to first get a general approval on the structure and code.

Reply:
I tried removing `anyhow` and there wasn't a clean way to provide the same context to errors without replicating much of the machinery anyhow provides. So I am going to leave it for now.

Reply:
What I have done though, is hiding anyhow externally.